### ISSUE TITLE
Depending on the number value of custom claim, DefaultClaims#get() returns Integer or Long

### ISSUE THREAD

**Issue by sainaen** [jwtk/jjwt#142]
Simple test case:
```groovy
  String claimName = "long"
  Long claimVal = new Long(5)
  String compact = Jwts.builder().setClaims([(claimName): claimVal]).compact();
  Claims claims = Jwts.parser().parse(compact).body as Claims
  // `claims.get` throws `RequiredTypeException`
  assert claims.get(claimName, Long) == claimVal
```
Here, Jackson finds that a `5` is small enough to fit in the `Integer` and so unmarshalls into it, instead of expected `Long`.

What makes matters worse is that when a value is large enough (`≥ 2^31`), Jackson will switch to `Long` and so code that naively always `get()`-s `Integer` will stop working:
```groovy
  String claimName = "long"
  Long claimVal = new Long((Integer.MAX_VALUE as long) + 5)
  String compact = Jwts.builder().setClaims([(claimName): claimVal]).compact();
  Claims claims = Jwts.parser().parse(compact).body as Claims
  // ok
  assert claims.get(claimName, Long) == claimVal
  // throws
  assert claims.get(claimName, Integer) != null
```

I was going to submit a pull request with addition of simple `if` in the `DefaultClaims#get`, but than thought about all those similar use cases with `Byte`-s, `Short`-s, `Float`-s and `Double`-s, started implementing them, got a bit overwhelmed and decided to ask here if it's even worth going into such deeps or an `if` would be enough. What do you think?

**Comment by lhazlewood** [comment]
I'm open to ideas!  I think the best recourse here is to use the various `valueOf` methods if the current value is not of the expected type, for example:

```java
if (requiredType.isInstance(value)) {
  return requiredType.cast(value);
}
//otherwise need type conversion:
String sval = String.valueOf(value);
//use reflection to invoke Long.valueOf or Integer.valueOf, etc depending on what requiredType is
```

This isn't likely the optimal solution, but I hope the logic makes sense enough as a starting point.

**Comment by sainaen** [comment]
@masudias 
As far as I can tell, fix is included since v0.8.0.
